cloud/scope/powervs_cluster_test.go

```diff
@@ -3043,3 +3045,728 @@ func TestCreateVPCSubnet(t *testing.T) {
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeTrue())
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("When PowerVS service Instance and VPC details are not set", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			TransitGatewayClient: mockTransitGateway,
+			IBMVPCClient:         mockVPC,
+			ResourceClient:       mockResourceController,
+			IBMPowerVSCluster:    &infrav1beta2.IBMPowerVSCluster{},
+		}
+
+		mockTransitGateway.EXPECT().GetTransitGatewayByName(gomock.Any()).Return(nil, nil)
+		requeue, err := clusterScope.ReconcileTransitGateway()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+}
+
+func TestCheckAndUpdateTransitGatewayConnections(t *testing.T) {
+	var (
+		mockResourceController *mockRC.MockResourceController
+		mockVPC                *mock.MockVpc
+		mockTransitGateway     *mockTG.MockTransitGateway
+		mockCtrl               *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockTransitGateway = mockTG.NewMockTransitGateway(mockCtrl)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+		mockResourceController = mockRC.NewMockResourceController(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("Returns error when getting VPC details", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+		clusterScope := makePowerVSClusterScope(mockTransitGateway, mockVPC, mockResourceController)
+
+		mockTransitGateway.EXPECT().ListTransitGatewayConnections(gomock.Any()).Return(&tgapiv1.TransitGatewayConnectionCollection{}, nil, nil)
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(nil, nil, errors.New("failed to get vpc"))
+		requeue, err := clusterScope.checkAndUpdateTransitGatewayConnections(&tgapiv1.TransitGateway{ID: ptr.To("transitGatewayID"), Name: ptr.To("transitGatewayName")})
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("Returns error when getting PowerVS service Instance details", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+		clusterScope := makePowerVSClusterScope(mockTransitGateway, mockVPC, mockResourceController)
+
+		mockTransitGateway.EXPECT().ListTransitGatewayConnections(gomock.Any()).Return(&tgapiv1.TransitGatewayConnectionCollection{}, nil, nil)
+		mockVPC.EXPECT().GetVPC(gomock.Any()).Return(&vpcv1.VPC{CRN: ptr.To("crn")}, nil, nil)
+		mockResourceController.EXPECT().GetResourceInstance(gomock.Any()).Return(nil, nil, errors.New("failed to get serviceInstance"))
+		requeue, err := clusterScope.checkAndUpdateTransitGatewayConnections(&tgapiv1.TransitGateway{ID: ptr.To("transitGatewayID"), Name: ptr.To("transitGatewayName")})
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("TransitGateway connections doesn't exits, creates connections", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("When TransitGateway connections doesn't exist and creates connections", func(t *testing.T) {
```
Please change the format wherever applicable